### ISSUE TITLE
Dropdowns in FlxG.debugger.track / FlxG.debugger.wtach

### ISSUE THREAD

**Issue by mastef** [HaxeFlixel/flixel#1924]
It would be very nice to have some additional functionality in the FlxG debugger, such as dropdowns with enums on certain values. So instead of having to type `true` / `false`, you'd simply have a dropdown where you can pick the value.

This would also allow a bit nicer functionalities that could help with level editors, as you could simply pick and choose from pre-defined values for an Object, without having to reinvent the wheel.

A nice show-case of where such a live-feature works, is e.g in the Chrome inspector. Select some element, clear a css value with a pre-defined word, and it'll auto suggest all the different values that you can pick : 
![image](https://cloud.githubusercontent.com/assets/762297/18165007/197cdf06-706d-11e6-9c7d-ad3c740d6206.png)

**Comment by MSGhero** [comment]
So is this filtering the completion window suggestions for certain class types?

**Comment by Gama11** [comment]
No, it's a dropdown that's provided for Watch entries where the type of the variable being edited is some sort of enumeration (boolean, an enum, not sure if there are others).

**Comment by mastef** [comment]
Example : You're watching a boolean, e.g. camera.active and it's set to `false`. You click on it, and you have the choices : `false, true`. You click on `true` and the state of camera.active switches to true.

Another example : 

You have `level_id` set to `my_level1`. You click on it, and a callback provides you with a list of possible values. From the list you pick `my_level3` and have switched your level to another one with 2 clicks.

It gets pretty useful quickly. Right now the lack of copy/paste and the freeform typing for different kinds of types are a bit of a bummer. Drop downs providing possible options ( through enums, or callbacks ) could be nice

**Comment by Gama11** [comment]
You can now cycle through Bool values by pressing the up / down keys while editing the value of a watch entry. We could probably allow the same for enum values.

**Comment by MSGhero** [comment]
Question:

If the value is actually coming from an hscript expression, what happens with the editable field? Like, if I'm tracking `sprite.x / 20` then I click the field and hit up or down, what happens? If it's a Boolean, will it suggest true/false to me?

I'm on mobile atm so I can't check myself.

**Comment by Gama11** [comment]
Alright, this now works for enum values too. :) Would maybe still be nice to have a dropdown / "dropup" alternatively.

@MSGhero Only `WatchEntryData.FIELD` entires are editable.
